### PR TITLE
Write examples of customizing `Tracer`

### PR DESCRIPTION
**Pull request by ansley**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#51668 Write examples of customizing `Tracer`**

### CODE REVIEW

File: torch/fx/symbolic_trace.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ class Tracer(TracerBase):
     """
     ``Tracer`` is the class that implements the symbolic tracing functionality
     of ``torch.fx.symbolic_trace``. A call to ``symbolic_trace(m)`` is equivalent
-    to ``Tracer().trace(m)``.
+    to ``Tracer().trace(m).graph``.
```
Comment (jamesr66a) [inline torch/fx/symbolic_trace.py:55]: `Tracer().trace(m)` returns a `Graph`, not GraphModule`. I actually think this docstring should say something like `symbolic_trace(m) is equivalent to GraphModule(m, Tracer().trace(m))`

File: torch/fx/examples/custom_tracer.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+"""
+How to Create and Use Custom Tracers
+
+`Tracer`--the class that implements the symbolic tracing functionality
+of `torch.fx.symbolic_trace`--can be subclassed to override various
+behaviors of the tracing process. In this tutorial, we'll demonstrate
+how to customize the symbolic tracing process using some handwritten
+Tracers. Each example will show that, by simply overriding a few methods
+in the `Tracer` class, you can completely alter the Graph produced by
```
Comment (jamesr66a) [inline torch/fx/examples/custom_tracer.py:13]: Not sure if completely altering it is a selling point, I would just say "you can alter the Graph"

"Completely alter" would necessarily bring up questions of "so is tracing not generally usable downstream?"

File: torch/fx/examples/custom_tracer.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+symbolic tracing. For a complete description of the methods that
+can be changed, refer to the docstrings of the methods in the Tracer
+class.
```
Comment (jamesr66a) [inline torch/fx/examples/custom_tracer.py:16]: Maybe add a link to the docs (can just be the master branch docs right now)

File: torch/fx/examples/custom_tracer.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+Note that a call to `symbolic_trace(m)` is equivalent to
+`Tracer().trace(m).graph`. (`Tracer` is the default implementation of
+Tracer as defined in `symbolic_trace.py`.)
+"""
+
+
+
+
+
```
Comment (jamesr66a) [inline torch/fx/examples/custom_tracer.py:31]: nit: Seems like a lot of blank lines

File: torch/fx/examples/custom_tracer.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+class.
+
+If you want a real-world example of a custom tracer, check out FX's AST
+Rewriter in `rewriter.py`. `RewritingTracer` inherits from Tracer but
+overrides the `trace` function so that we can rewrite all calls to
+`assert` to the more FX-friendly `torch.assert`.
+
+Note that a call to `symbolic_trace(m)` is equivalent to
+`Tracer().trace(m).graph`. (`Tracer` is the default implementation of
+Tracer as defined in `symbolic_trace.py`.)
```
Comment (jamesr66a) [inline torch/fx/examples/custom_tracer.py:25]: Same comment as below, it's the other way around. `symbolic_trace` returns a `GraphModule` but `Tracer.trace` returns a `Graph`

File: torch/fx/examples/custom_tracer.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+class M1(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.relu = torch.nn.ReLU()
+
+    def forward(self, x):
+        return self.relu(x)
+
+"""
```
Comment (jamesr66a) [inline torch/fx/examples/custom_tracer.py:53]: Add the code that creates this printout

**Review by jamesr66a — COMMENTED** [review]
Nice!

**Comment by jamesr66a** [comment]
Hey can you move this to a PR against the `examples` repo? https://github.com/pytorch/examples/tree/master/fx

Thanks!

**Comment by ansley** [comment]
Made a PR to `pytorch/examples` instead

**Pull request closed without merging**